chore: windows winget package auto update on every release

matifali opened the pull request:
Added `winget` publish action on release.
Linked to #3714
@deansheather

matifali committed:
Create submit-windows-packages.yml

Added `winget` update action on release

matifali commented:
Not sure how to test it. Will it submit the package in my name or the coder

matifali committed:
Rename submit-windows-packages.yml to submit-windows-packages.yaml

matifali committed:
Merge branch 'coder:main' into winget-release

matifali committed:
Added GITHUB_PAT Token 

A GitHub Personal Access token is required to submit packages to winget community repo.

https://github.com/microsoft/winget-create/blob/main/doc/token.md

Please create a token as shown here and store it as a secret in `GITHUB_PAT`

matifali commented:
To push an update, we first need to create a package for the first time that is interactive and manual.
I am not sure if it has to be created using Coder Inc. GitHub or if I can do it from my account.
Meanwhile, I have created a PR  here also,
https://github.com/microsoft/winget-pkgs/pull/86416
and tried to fill in all the required metadata from coder.com/

matifali commented:
Also,
A GitHub Personal Access token is required to submit packages to winget community repo.

https://github.com/microsoft/winget-create/blob/main/doc/token.md

Please create a token as shown here and store it as a secret in `GITHUB_PAT`

matifali commented:
My PR was closed as a dublicate of #

deansheather commented:
I'll do a review pass of this PR on Monday and get it merged for you. We'll sort out the bounty payout post-merge cc @ghuntley

matifali committed:
Merge branch 'coder:main' into winget-release

deansheather committed:
chore: use gh cli in winget automation, comment on PR

deansheather commented:
I updated your PR to use GH CLI instead of pinging the API directly (to avoid potential ratelimits), as well as updated it to comment on the PR pinging me so microsoft maintainers can get in contact with me if something's wrong.

I also discovered that the file version I was inserting into the installer binary was incorrect so I fixed that as well.

deansheather reviewed (commented):
LGTM but I contributed so I'll get Colin to review it

matifali commented:
> I updated your PR to use GH CLI instead of pinging the API directly (to avoid potential ratelimits), as well as updated it to comment on the PR pinging me so microsoft maintainers can get in contact with me if something's wrong.
> 
> I also discovered that the file version I was inserting into the installer binary was incorrect so I fixed that as well.

yes, it looks like a better approach. Yes, the installer version had the 4th section as the release date.

matifali commented on .github/workflows/packages.yaml:
```diff
@@ -0,0 +1,51 @@
+        id: version
+        run: |
+          $version = $env:CODER_VERSION -replace "^v", ""
+          $version += ".0"
```
Why are you using a trailing 0? Usually, the coder releases have 3-digit version numbers.

deansheather replied:
The NSIS installer required me to add the extra 4th section to the version number, so it shows up in Windows settings app as `0.11.0.x`. Not sure if that's a Windows requirement or an NSIS requirement, but for consistency between winget and the installed package on Windows I opted to keep these the same.

I changed the windows installer build script to add `.0` instead of the date as the date was actually being clipped off funnily enough as it's not a 16 bit number

ghuntley commented:
Thanks for sending in the pull-request @matifali I'm going to have a look/review after lunch. A quick note to say haven't forgotton!

matifali committed:
Merge branch 'coder:main' into winget-release
